From a2d342ddb4b6fd1f7e704de322306de4e0f78d80 Mon Sep 17 00:00:00 2001
From: Ravishankar N <ravishankar@redhat.com>
Date: Fri, 18 Dec 2015 15:22:35 +0530
Subject: [PATCH 194/214] afr: Fix bug in afr_inode_refresh_do

Introduced by commit 7e3550c0ad4dfc197e5cbfd38748f5eab75bcf94
(https://code.engineering.redhat.com/gerrit/#/c/64009/)
downstream.

The original upstream patch http://review.gluster.org/#/c/12894/ has the
fix incorporated.

Change-Id: Ia54b709f3bd372deefc4142005fc9efdebebe0b3
BUG: 1275751
Signed-off-by: Ravishankar N <ravishankar@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/64156
Reviewed-by: Pranith Kumar Karampuri <pkarampu@redhat.com>
Tested-by: Pranith Kumar Karampuri <pkarampu@redhat.com>
---
 xlators/cluster/afr/src/afr-common.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xlators/cluster/afr/src/afr-common.c b/xlators/cluster/afr/src/afr-common.c
index 6c75cc9..0d0270a 100644
--- a/xlators/cluster/afr/src/afr-common.c
+++ b/xlators/cluster/afr/src/afr-common.c
@@ -984,7 +984,7 @@ afr_inode_refresh_do (call_frame_t *frame, xlator_t *this)
                 for (i = 0; i < priv->child_count; i++) {
                         if (local->child_up[i] &&
                             fd_ctx->opened_on[i] == AFR_FD_OPENED)
-                                wind_subvols[1] = 1;
+                                wind_subvols[i] = 1;
                 }
         } else {
                 memcpy (wind_subvols, local->child_up,
-- 
1.7.1

